### DISCUSSION TITLE
Iceberg Glue cannot find catalog during CREATE TABLE AS

### DISCUSSION THREAD

**Discussion by olivier-derom** [trinodb/trino#23640] [General]
I encountered this issue using dbt.

I have a fresh glue catalog in Trino.
I run my dbt job.
dbt automatically creates the schema in the catalog, and afterwards tries to run 

```
create table "iceberg-glue"."myschema"."execution_time"
    as (
    	SELECT
    	current_timestamp(3) AS execution_time
    )
```

I get error `SQL Error [44]: Query failed (#20241001_142340_00096_8hyhs): No catalog 'iceberg-glue'`
But I can verify in DBeaver it is there, and the schema in the catalog also has been created.
If i try to run the same query manually in a console, I get the same Trino error.

When I replace CREATE TABLE with CREATE VIEW, it works, the view gets created, no errors.
If I try to CREATE TABLE like
`create table "iceberg-glue"."myschema"."execution_time" (execution_time VARCHAR)`
it also works.

But even when I have a table and a view live in the database, the CREATE TABLE AS ... will throw 'catalog not found', no matter what I try.

This seems to be specific to the glue catalog, as we were previously using rest catalog, and it had no issues with this.

my catalog setup:
```
iceberg-glue: |
      connector.name=iceberg
      iceberg.catalog.type=glue
      iceberg.file-format=PARQUET
      iceberg.compression-codec=SNAPPY
      hive.metastore.glue.region=eu-west-1
      hive.metastore.glue.default-warehouse-dir=s3://somebucket/iceberg-glue/
      hive.metastore.glue.catalogid=123456789000
      hive.s3.region=eu-west-1
```

Let me know if you need any more information

**Comment by ebyhr** [comment]
The issue usually happens when the catalog config property isn't deployed to all workers. Could you confirm that?

**Comment by olivier-derom** [comment]
@ebyhr Thanks for the hint!
We are deploying using the helm chart, the coordinator automatically redeploys when a change has been picked up in the catalog, but it seems the workers don't, which was indeed needed. Manually redeploying the workers fixed the issue.

Thanks as always!

**Reply by nineinchnick** [reply]
That's fixed in the 0.29.0 version of the chart (one before the latest): https://github.com/trinodb/charts/releases/tag/trino-0.29.0

**Reply by olivier-derom** [reply]
We were on 0.28.0, thanks for letting me know!
